openid-4-verifiable-presentations-1_0.md

```diff
@@ -215,6 +217,55 @@ Figure: Cross Device Flow
 
 (3) The Wallet prepares the Verifiable Presentation(s) of the Verifiable Credential(s) that the End-User has consented to. It then sends to the Verifier an Authorization Response where the Verifiable Presentation(s) are contained in the `vp_token` parameter.
 
+## Same Device Flow with Wallet Capabilities Discovery {#same_device_with_discovery}
+
+Below is a diagram that shows the flow as described in (#same_device), but with an additional pre-step used that allows the Verifier to discover the Wallet's capabilities and tailor the actual request.
+
+In this case, the Verifier starts the flow by discovering the Wallet's capabilities through an additional message exchange. This is especially useful, if the Verifier starts the process using a URL serving as alias for a group of wallets (e.g. a custom scheme) but would like to tailor the presentation request to the particular wallet. 
```

Comment:
We should add the following Note somewhere
Note. In order to prevent a Verifier from fingerprinting the wallet, it is strongly recommended that one or more standardised sets of capabilities are defined that may be commonly supported by wallets. This will allow the Verifier to request conformance to one or more sets, and the wallet to return an assertion about one or more sets, thereby protecting the privacy of the wallet implementation.

Reply:
I agree, since there may be cases where the wallet instances doesn't support discovery, then the RP has to rely on a standard set. 

@David-Chadwick do you think htat this set should be recommended in the current spec or that it would be out of scope and then it should be defined within each impl profile? (eg: EUDI Wallet should have its standard set ... and so on)

Reply:
Since David is not concerned about interoperability here, but privacy, this could go into the privacy considerations I mentioned in my comment on this PR. However, I think that just using some set of default capabilities is not sufficient: There are many, many ways in which information about the wallet can be leaked. Browser vendors are working quite hard to make browser fingerprinting harder, and we need to take a page or two out of their book if we allow this request to happen without user consent, to implement similar protections. For example: Just the presence of a particular wallet might leak the nationality of a user, and that particular wallet might send parameters in a different order than all other wallets. Or: The wallet might support a different version of HTTP than all other wallets.

Reply:
I agree that privacy protection is extremely difficult (and maybe its impossible for it to be perfect) but we should at least add guidelines or recommendations into how the capabilities detection feature should be used/implemented.
I am happy for my proposed text to go into the privacy considerations section, supplemented with additional cautionary text from Daniel.
@peppelinux Certainly profiles of the spec will define their own set of required capabilities for all implementors of the profile, but if the WG determine that there is a useful common subset that is widely applicable, this could go into an informative annex. (An analogy is the X.500 spec that defined a common set of useful attributes that most people implemented, whilst acknowledging that other groups of users would define their own additional attributes.)

Reply:
@danielfett by tracking Wallet Instance IPv4 with GeoIP it would be enough to track nationality and also mobile broadband provider

Reply:
@danielfett even if a default capabilities is not sufficient (I agree) specs like OIDC requires the sigalg RS256 as a strong recommendation to the entities, SAML2 made something similar in the real-world with sha-1 and then SHA256. I mentions these just to continuing debating on the requirement of giving more guidance to implementers

Reply:
Please suggest text that I can add to the privacy considerations section.